src/main/java/emissary/core/BaseDataObject.java

```diff
@@ -273,6 +279,20 @@ public void setFilename(final String f) {
         return this.theData;
     }
 
+    /**
+     * Return a String version of the data. If there is already a cached version of the dataString, then return that,
+     * otherwise generate the String to be returned and cache the result
+     *
+     * @return a String representation of the data
+     */
+    @Override
+    public String dataAsString() {
+        if (this.dataString == null) {
+            this.dataString = new String(this.theData, StandardCharsets.UTF_8);
```

Comment:
It is safe to assume that the bytes will always be UTF-8? There may be cases where the data in byte[] is a non UTF-8 string or not even a string. It would be nice to provide a mechanism to pass a character encoding along in `setData()` if the caller know what that is -  or even provide an alternate to `setData()` that takes a `String` as a parameter.

Reply:
Additionally, there should probably be a null check on "this.theData".